tests/components/tplink/test_switch.py

```diff
@@ -52,6 +58,18 @@ async def test_plug(hass: HomeAssistant) -> None:
     plug.turn_on.assert_called_once()
     plug.turn_on.reset_mock()
 
+    await hass.services.async_call(
+        SWITCH_DOMAIN, "turn_off", {ATTR_ENTITY_ID: led_entity_id}, blocking=True
+    )
+    plug.set_led.assert_called_once_with(False)
+    plug.set_led.reset_mock()
+
+    await hass.services.async_call(
+        SWITCH_DOMAIN, "turn_on", {ATTR_ENTITY_ID: led_entity_id}, blocking=True
+    )
+    plug.set_led.assert_called_once_with(True)
+    plug.set_led.reset_mock()
```

Comment:
Could you move the led specific tests into a separate test function?

Reply:
Sure, I'll do it!